system/jlib/jptree.cpp

```diff
@@ -3046,6 +3064,66 @@ void PTree::serializeToStream(IBufferedSerialOutputStream &tgt) const
     serializeCutOff(tgt, -1, 0);
 }
 
+void PTree::deserializeFromStream(IBufferedSerialInputStream &src)
+{
+    deserializeSelf(src);
+
+    StringBuffer eName;
+    for (;;)
+    {
+        eName.clear();
+        peekReadZeroTerminatedString(src, eName);
+        if (eName.isEmpty())
+        {
+            src.skip(1); // Skip over null terminator.
+            break;
+        }
+        IPropertyTree *child = create(src);
```

Comment:
Future optimization: Don't read the name - just check if it is blank, call create() and then pass child->queryName() to addPropTree.